### PR TITLE
fix: only load editor once even if connection is recovered

### PR DESCRIPTION
**Pull request by max-nextcloud**
### 📝 Summary

Only load the editor if it has not been loaded yet.
~~Move `createEditor` out of the `onLoaded` function.~~

`onLoaded` may be called multiple times.
When the connection to the server is lost and restored the collaboration extension reloads the websocket polyfill which in turn opens the SyncService
which will trigger the `onCreated` and `onLoaded` callbacks.

* Resolves: #3785 

### 🚧 TODO

When loosing the connection the yjs websocket provider will try to reconnect automatically
However we still show the reconnect button. The automatic reconnection succeeded for me - but the button fails.

- [ ] Hide the reconnect button in that state
- [ ] Properly disconnect after a timeout and show reconnect dialogue then
- [ ] Make sure the reconnect actually works

### 🏁 Checklist

- [x] Code is properly formatted (`npm run lint` / `npm run stylelint` / `composer run cs:check`)
- [x] [Sign-off message](https://probot.github.io/apps/dco/) is added to all commits
- [x] Documentation is not required
- [ ] [Tests](https://github.com/nextcloud/text#-testing-the-app) (unit, integration and/or end-to-end) passing and the changes are covered with tests

### CODE REVIEW

**Comment by max-nextcloud** [comment]
This PR includes a test that cancels some requests to simulate a connection loss.
I was able to trigger the same behavior locally by stopping the nextcloud-docker-dev environment and restarting it a few seconds later. That way it's probably easier to explore this in the browser oneself.

**Comment by max-nextcloud** [comment]
We will need to work on the TODO items listed above. However I think it would still be good to land this today before the RC to mitigate some of the problems we are seeing on the server right now.

File: src/services/SyncService.js [hunk 1]
```diff
@@ -98,8 +98,8 @@ class SyncService {
 
 		// TODO: Only continue if a connection was made
 		this.connection = await connect
-
-		this.version = this.connection.docStateVersion
+		this.backend = new PollingBackend(this, this.connection)
+		this.version = this.connection.lastSavedVersion
```
Comment (juliushaertl) [inline src/services/SyncService.js:102]: Was this reverted on purpose from https://github.com/nextcloud/text/pull/3842/files#diff-103df33bb085aaec0d6134982c312e96ecf58cf09335b9cf15964e17a45b4071 ?

Comment (max-nextcloud) [reply src/services/SyncService.js:102]: good catch... no that's left from the rebase!

Comment (max-nextcloud) [reply src/services/SyncService.js]: good catch... no that's left from the rebase!

**Comment by max-nextcloud** [comment]
Okay... found a **much** simpler way of basically achieving the same.

One issue remains though:
After the automatic session recovery the old session will still be used for the mention extension.
Therefore mentions will not work anymore.

**Review by juliushaertl — APPROVED** [review]

**Review by mejo- — APPROVED** [review]
Code looks good to me :ok_hand:

I don't fully understand the impact of moving up the initialization of `this.backend` above the emits.

**Pull request merged**